src/test/java/com/sonyericsson/rebuild/RebuildValidatorTest.java

```diff
@@ -393,6 +398,110 @@ public void testRebuildSupportedUnknownParameterValue() throws Exception {
+				"1/rebuild");
+		// Rebuild (#2)
+		submit(rebuildConfigPage.getFormByName("config"));
+
+		HtmlPage projectPage = createWebClient().getPage(project);
+		WebAssert.assertLinkPresentWithText(projectPage, "Rebuild Last");
+
+		HtmlAnchor rebuildHref = projectPage.getAnchorByText("Rebuild Last");
+		assertEquals("Rebuild Last should point to the second build", "/"
+						+ project.getUrl() + "lastCompletedBuild/rebuild",
+				rebuildHref.getHrefAttribute());
+
+		List<ParameterValue> parameterValues = project.getLastCompletedBuild().getAction(ParametersAction.class).getParameters();
+
+		assertThat(parameterValues, hasStringParamValues(equalToStringParamValue("existing", "overridingValue")));
+		assertThat(parameterValues, hasNoStringParamValues(equalToStringParamValue("existing", "defaultValue")));
+	}
+
+	public void testWhenProjectWithExistingParamAndInjectedParamWhenRebuild()
+			throws Exception {
+		FreeStyleProject project = createFreeStyleProject();
+		project.addProperty(new ParametersDefinitionProperty(
+				new StringParameterDefinition("oldName", "oldValue")));
+
+		// Build (#1)
+		project.scheduleBuild2(0, new Cause.UserIdCause(),
+				new ParametersAction(new StringParameterValue("injectedName", "injectedValue")))
+				.get();
+		HtmlPage rebuildConfigPage = createWebClient().getPage(project,
+				"1/rebuild");
+		// Rebuild (#2)
+		submit(rebuildConfigPage.getFormByName("config"));
+
+		List<ParameterValue> parameterValues = project.getBuildByNumber(2).getAction(ParametersAction.class).getParameters();
+
+		assertThat(parameterValues, hasStringParamValues(
+				equalToStringParamValue("oldName", "oldValue"),
+				equalToStringParamValue("injectedName", "injectedValue")
+		));
+	}
+
+	public void testWhenProjectWithExistingParamAndInjectedParamWhenRebuildOfLastBuild()
+			throws Exception {
+		FreeStyleProject project = createFreeStyleProject();
+		project.addProperty(new ParametersDefinitionProperty(
+				new StringParameterDefinition("oldName", "oldValue")));
+
+		// Build (#1)
+		project.scheduleBuild2(0, new Cause.UserIdCause(),
+				new ParametersAction(new StringParameterValue("injectedName", "injectedValue")))
+				.get();
+		HtmlPage rebuildConfigPage = createWebClient().getPage(project,
+				"1/rebuild");
+		// Rebuild (#2)
+		submit(rebuildConfigPage.getFormByName("config"));
+
+		HtmlPage projectPage = createWebClient().getPage(project);
+		WebAssert.assertLinkPresentWithText(projectPage, "Rebuild Last");
+
+		HtmlAnchor rebuildHref = projectPage.getAnchorByText("Rebuild Last");
+		assertEquals("Rebuild Last should point to the second build", "/"
+						+ project.getUrl() + "lastCompletedBuild/rebuild",
+				rebuildHref.getHrefAttribute());
+
+		List<ParameterValue> parameterValues = project.getLastCompletedBuild().getAction(ParametersAction.class).getParameters();
+
+		assertThat(parameterValues, hasStringParamValues(
+				equalToStringParamValue("oldName", "oldValue"),
```

Comment:
you're just calling the constructor, you should just be doing `new StringParameterValue`